### PR TITLE
[fix])(nereids)Support select catalog.db.table.column from xxx for nereids planner.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Nereids doesn't support `select table.* from table`, this pr is to fix this bug.
Support three layer qualifier. (catalog.database.table)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.65 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162015966 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalCatalogRelation.java [hunk 1]
```diff
@@ -97,13 +98,19 @@ public List<String> getQualifier() {
...
+            return qualifier;
+        }
         return Utils.qualifiedNameParts(qualifier, table.getName());
     }
 
     /**
      * Full qualified table name, concat qualifier and name with `.` as separator.
      */
     public String qualifiedName() {
+        if (qualifier.size() == 3) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalCatalogRelation.java:111]: why we use special branch?

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalCatalogRelation.java (hunk 1 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalCatalogRelation.java:111]: qualifier may only contain db name or contain full name (catalog.db.table)，for the first case, need to concat the table name here, for the latter case, we don't need to.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**